lib/segment/src/index/field_index/map_index/mod.rs

```diff
@@ -482,7 +575,11 @@ impl PayloadFieldIndex for MapIndex<IntPayloadType> {
     }
 
     fn clear(self) -> OperationResult<()> {
-        self.get_db_wrapper().recreate_column_family()
+        match self {
+            MapIndex::Mutable(index) => index.get_db_wrapper().recreate_column_family(),
+            MapIndex::Immutable(index) => index.get_db_wrapper().recreate_column_family(),
+            MapIndex::Mmap(index) => index.clear(),
+        }
```

Comment:
Nit: it could be deduplicated into a common code for both `str` and `IntPayloadType`. Same for remove_point.

Reply:
`clear()`:
- https://github.com/qdrant/qdrant/blob/63f6cd3dbec8f18942cf9d7ae6ba9d9f47a1503b/lib/segment/src/index/field_index/map_index/mod.rs#L463-L469
- https://github.com/qdrant/qdrant/blob/63f6cd3dbec8f18942cf9d7ae6ba9d9f47a1503b/lib/segment/src/index/field_index/map_index/mod.rs#L577-L583

`remove_point()`:
- https://github.com/qdrant/qdrant/blob/63f6cd3dbec8f18942cf9d7ae6ba9d9f47a1503b/lib/segment/src/index/field_index/map_index/mod.rs#L704-L713
- https://github.com/qdrant/qdrant/blob/63f6cd3dbec8f18942cf9d7ae6ba9d9f47a1503b/lib/segment/src/index/field_index/map_index/mod.rs#L742-L751

I suggest to add these two methods to `impl<N: MapIndexKey + MmapValue + ?Sized> MapIndex<N>`:
https://github.com/qdrant/qdrant/blob/63f6cd3dbec8f18942cf9d7ae6ba9d9f47a1503b/lib/segment/src/index/field_index/map_index/mod.rs#L81

Reply:
Thanks! Got it, done